src/inflators/loop-animation.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  addComponent,
+  hasComponent
+} from "bitecs";
+import {
+  AnimationAction,
+  AnimationClip,
+  AnimationMixer,
+  LoopRepeat
+} from "three";
+import {
+  LoopAnimation,
+  LoopAnimationData,
+  MixerAnimatable,
+  MixerAnimatableData,
+  Object3DTag
+} from "../bit-components";
+import { HubsWorld } from "../app";
+
+// TODO: Type definition should be done in bit-components?
+type MixerAnimatableData = Map<number, {
+  mixer: AnimationMixer,
+  animations: Array<AnimationClip>
+}>;
+type LoopAnimationSata = Map<number, Array<AnimationAction>>;
+
+export type LoopAnimationParams = {
+  activeClipIndex: number;
+  paused: boolean;
+  startOffset: number;
+  timeScale: number;
+  // TODO: Do we need to keep supporting the following two params?
+  // DEPRECATED: Use activeClipIndex instead since animation
+  //             names are not unique
+  clip?: string;
+  // Support for Spoke->Hubs activeClipIndices struct
+  activeClipIndices?: Array<number>;
+};
+
+const DEFAULTS = {
+  activeClipIndex: 0,
+  paused: false,
+  startOffset: 0,
+  timeScale: 1.0
+};
+
+const getActiveClips = (animations: Array<AnimationClip>, params: LoopAnimationParams) : Array<AnimationClip> => {
+  if (params.activeClipIndices !== undefined &&
+      params.activeClipIndices.length > 0) {
+    return params.activeClipIndices.map(index => animations[index]);
+  } else if (params.clip !== undefined &&
+             params.clip !== "") {
+    const activeClips = [];
+    const clipNames = params.clip!.split(",");
+    for (let i = 0; i < clipNames.length; i++) {
+      const clipName = clipNames[i];
+      const foundClip = animations.find((clip: AnimationClip) => {
```

Comment:
Unfortunately since we need to support named animation lookup we will need to pass that all the way through to the system where we have the enterQuery since we can't really look up the animations here.